drivers/gpio/gpio_nrfx.c

```diff
@@ -328,19 +316,36 @@ static int gpio_nrfx_pin_interrupt_configure(struct device *port,
 		return -ENOTSUP;
 	}
 
-	WRITE_BIT(data->pin_int_en, pin, flags & GPIO_INT_ENABLE);
-	WRITE_BIT(data->int_en, pin, true);
-	WRITE_BIT(data->trig_edge, pin, flags & GPIO_INT_EDGE);
-	WRITE_BIT(data->double_edge, pin, (flags & GPIO_INT_LOW_0) &&
-					  (flags & GPIO_INT_HIGH_1));
-
-	bool active_high = ((flags & GPIO_INT_HIGH_1) != 0);
+	bool pin_int_en = true;
+	bool edge = true;
+	bool active = true;
 
-	if (((flags & GPIO_INT_LEVELS_LOGICAL) != 0) &&
-	    ((data->general.invert & BIT(pin)) != 0)) {
-		active_high = !active_high;
+	switch (flags) {
+	case GPIO_INT_MODE_DISABLED:
+		pin_int_en = false;
+		break;
+	case GPIO_INT_MODE_LEVEL_LOW:
+		edge = false;
+		active = false;
+		break;
+	case GPIO_INT_MODE_LEVEL_HIGH:
+		edge = false;
+		break;
+	case GPIO_INT_MODE_EDGE_FALLING:
+		active = false;
+		break;
+	case GPIO_INT_MODE_EDGE_RISING:
+		break;
+	case GPIO_INT_MODE_EDGE_BOTH:
+		edge = true;
+		WRITE_BIT(data->double_edge, pin, true);
```

Comment:
The `gpio_api_1pin_test` didn't catch it but this looks like an error. If we run `gpio_pin_interrupt_configure` to enable interrupt on both edges and then re run the function on the same pin to enable interrupt on the rising edge only the `data->double_edge` will be stuck at `1` indicating double edge interrupt.